### DISCUSSION TITLE
diamond error! error code: -9

### DISCUSSION THREAD

**Discussion by fnm2225** [oschwengers/bakta#185] [Help wanted]
Hello, I was wondering if someone could help me. I´ve recently installed bakta via docker and when I use it, the process is killed and the following message pops up: 

Traceback (most recent call last):
  File "/opt/conda/bin/bakta", line 8, in <module>
    sys.exit(main())
  File "/opt/conda/lib/python3.8/site-packages/bakta/main.py", line 249, in main
    cdss_psc, cdss_pscc, cdss_not_found = psc.search(cdss_not_found)
  File "/opt/conda/lib/python3.8/site-packages/bakta/psc.py", line 64, in search
    raise Exception(f'diamond error! error code: {proc.returncode}')
Exception: diamond error! error code: -9

**Comment by oschwengers** [comment]
Hi @fabrizziom99 ,
this might be a memory issue of Diamond. Could you try to execute the container with more memory? Normally 16 Gb is enough, but sometimes Diamond needs even more.

**Reply by fnm2225** [reply]
Indeed it was a memory issue, solved it by running bakta after upgrading my RAM. Thanks!!

**Reply by bio-Tao** [reply]
I see Diamond has option -b，
![image](https://github.com/oschwengers/bakta/assets/74087279/830ed0c9-2262-423a-8f94-c3c97e1c5cad)
can add it for lower memory users?

**Reply by oschwengers** [reply]
Hi, sry, for the very late reply. Currently, this is not the case. We could maybe think about an automated adjustment based on free memory. But I'm a bit scared of potential side effects.

**Reply by Pashane240** [reply]
Hi, I'm having the same problem. I want to know how much RAM did you install to solve the issue?

**Reply by oschwengers** [reply]
Hi. Diamond's memory requirements within Bakta highly depend on the size/complexity of the genome and how many genes are actually searched, i.e. could not be identified via the AFSI approach.

Sometimes, even 8 GB is enough, but usually, we use **16 GB**.
However, we regularly see a rare fraction of genomes for which we have to increase this limit to **32 GB**.
